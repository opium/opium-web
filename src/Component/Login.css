.Login__Container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
}

.Login {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    height: 100%;
    max-height: 300px;
}

.Login__InputContainer {
    width: 100%;
    max-width: 450px;
}

.Login__Input {
    width: 100%;
    max-width: 450px;
    padding: 5px 10px;
    margin-top: 5px;
}

.Login__Input--Submit {
    background: rgb(89, 79, 63);
    color: #fff;
    border: 0;
    padding: 10px;
    border-radius: 2px;
    cursor: pointer;
}

.Login__Input--Submit:hover, .Login__Input--Submit:active {
    background: rgb(79, 58, 27);
}
