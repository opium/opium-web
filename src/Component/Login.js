import React, { Component } from 'react';

class Login extends Component {
  constructor(props) {
    super(props);

    this.handleChangeLogin = this.handleChangeLogin.bind(this);
    this.handleChangePassword = this.handleChangePassword.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

    this.state = {
      login: '',
      password: '',
    };
  }

  handleChangeLogin(event) {
    this.setState({login: event.target.value});
  }

  handleChangePassword(event) {
    this.setState({password: event.target.value});
  }

  handleSubmit(event) {
    event.preventDefault();

    this.props.doLogin({
      login: this.state.login,
      password: this.state.password,
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type="text"
          placeholder="login"
          onChange={this.handleChangeLogin}
          value={this.state.login}
        />
        <input
          type="text"
          placeholder="password"
          onChange={this.handleChangePassword}
          value={this.state.password}
        />

        <button type="submit">
          Submit
        </button>
      </form>
    );
  }
}

export default Login;
