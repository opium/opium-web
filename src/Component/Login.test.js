import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

it('renders login', () => {
  const component = renderer.create(<Login
    doLogin={jest.fn()}
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});
