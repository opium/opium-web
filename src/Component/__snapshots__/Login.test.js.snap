exports[`test renders login 1`] = `
<div
  className="Login__Container">
  <form
    className="Login"
    onSubmit={[Function]}>
    <div>
      Demo credentials are:
      <ul>
        <li>
          <div>
            Basic user
          </div>
          <ul>
            <li>
              login: \"test\"
            </li>
            <li>
               password: \"test\"
            </li>
          </ul>
        </li>
        <li>
          <div>
            Admin user
          </div>
          <ul>
            <li>
              login: \"admin\"
            </li>
            <li>
               password: \"admin\"
            </li>
          </ul>
        </li>
      </ul>
    </div>
    <div
      className="Login__InputContainer">
      <label>
        <div>
          Login
        </div>
        <input
          className="Login__Input"
          onChange={[Function]}
          placeholder="login"
          type="text"
          value="" />
      </label>
    </div>
    <div
      className="Login__InputContainer">
      <label>
        <div>
          Password
        </div>
        <input
          className="Login__Input"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value="" />
      </label>
    </div>
    <button
      className="Login__Input Login__Input--Submit"
      type="submit">
      Sign in
    </button>
  </form>
</div>
`;
